backend/app/rest/api/ssl.go

```diff
@@ -25,25 +28,71 @@ const (
 
 // SSLConfig holds all ssl params for rest server
 type SSLConfig struct {
-	SSLMode sslMode
-	Cert    string
-	Key     string
-	Port    int
+	SSLMode      sslMode
+	Cert         string
+	Key          string
+	Port         int
+	ACMELocation string
 }
 
 // httpToHTTPSRouter creates new router which does redirect from http to https server
+// with default middlewares. Used in 'static' ssl mode.
 func (s *Rest) httpToHTTPSRouter() chi.Router {
 	log.Printf("[DEBUG] create https-to-http redirect routes")
 	router := chi.NewRouter()
 	router.Use(middleware.RealIP, Recoverer)
 	router.Use(middleware.Throttle(1000), middleware.Timeout(60*time.Second))
 
-	router.HandleFunc("/*", func(w http.ResponseWriter, r *http.Request) {
+	router.Handle("/*", s.redirectHandler())
+	return router
+}
+
+// httpChallengeRouter creates new router which perform ACME "http-01" challenge response
+// with default middlewares. This part is necessary to obtain certificate from LE.
+// If it receives not a acme challenge it performs redirect to https server.
+// Used in 'auto' ssl mode.
+func (s *Rest) httpChallengeRouter(m *autocert.Manager) chi.Router {
+	log.Printf("[DEBUG] create http-challenge routes")
+	router := chi.NewRouter()
+	router.Use(middleware.RealIP, Recoverer)
+	router.Use(middleware.Throttle(1000), middleware.Timeout(60*time.Second))
+
+	router.Handle("/*", m.HTTPHandler(s.redirectHandler()))
+	return router
+}
+
+func (s *Rest) redirectHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		newURL := s.RemarkURL + r.URL.Path
 		if r.URL.RawQuery != "" {
 			newURL += "?" + r.URL.RawQuery
 		}
 		http.Redirect(w, r, newURL, http.StatusTemporaryRedirect)
 	})
-	return router
+}
+
+func (s *Rest) makeAutocertManager() *autocert.Manager {
+	return &autocert.Manager{
+		Prompt:     autocert.AcceptTOS,
+		Cache:      autocert.DirCache(s.SSLConfig.ACMELocation),
+		HostPolicy: autocert.HostWhitelist(s.getRemarkHost()),
+		Email:      s.DataService.AdminStore.Email(""),
```

Comment:
this can be a problem for any other store. I.e. for the default (static store) empty siteID will work, but for mongo store it won't

Reply:
I don't think you can get a proper admin's email in this case, because it runs on the server level and in fact, this may not be the same admin email as fo sites. Pls note - the goal of admin email(s) is to send "delete me" requests, and those can be routed differently by different sites as it happens on rest level. For acme I would suggest a minor change:

1. Add a new SSL param, smth like `acme-email` to options
1. Make it empty by default
1. If defined in CLI/env just use it
1. if not set to:
   - admin email if `AdminGroup.Type == "shared"`
   - to admin@+`RemarkURL` domain

Reply:
fixed [here](https://github.com/umputun/remark/pull/217/commits/2c5a2921eefe5ac86be5ee52a8ab960de8272b37)